File: LeetCode/面试题17.11单词距离.cpp
class Solution {
public:
    int findClosest(vector<string>& words, string word1, string word2) {
        int ans = INT_MAX;
        for (int i = 0; i < words.size(); i++) {
            u_map[words[i]].emplace_back(i);
        }
        auto vec1 = u_map[word1];
        auto vec2 = u_map[word2];
        for (int i = 0; i < vec1.size(); i++) {
            for (int j = 0; j < vec2.size(); j++) {
                ans = min(ans, abs(vec2[j] - vec1[i]));
            }
        }
        return ans;
    }

private :
    unordered_map<string, vector<int>> u_map;
};

File: LeetCode/面试题17.11单词距离.rb
# @param {String[]} words
# @param {String} word1
# @param {String} word2
# @return {Integer}
def find_closest(words, word1, word2)
    hash_word = Hash.new
    ind1 = []
    ind2 = []
    temp = 100000
    (0...words.size).each do |x|
        hash_word[x] = words[x] 
    end
    hash_word.each_key do |x|
        if hash_word[x] == word1 then
            ind1 << x
        end
        if hash_word[x] == word2 then
            ind2 << x
        end
    end
    for i in ind1 do
        for j in ind2 do
            if (i - j).abs <= temp then
                temp = (i - j).abs
            end
        end
    end
    return temp
end